### PR TITLE
User interface documentation [13243]

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: docs/rst/user_manual/user_interface.rst [hunk 1]
```diff
@@ -0,0 +1,172 @@
+.. include:: ../exports/alias.include
+
+.. _user_manual_user_interface:
+
+##############
+User Interface
+##############
+
+The |ddsrouter| is an application that is executed from command line.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:9]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Source Dependency Libraries
+---------------------------
+
+The |ddsrouter| depends on Fast DDS ``fastrtps`` and ``fastcdr`` libraries.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:19]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+In order to correctly execute the Router, be sure that ``fastrtps`` and ``fastcdr`` are sourced.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:20]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+.. code-block: bash
+
+    source <path-to-fastdds-installation>/install/setup.bash
+
+.. note:
+
+    If Fast DDS has been installed in the system, these libraries would be sourced by default.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:28]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    Usage: Fast DDS Router
+    Connect different DDS networks via DDS through LAN or WAN.
+    It will build a communication bridge between the different Participants included in the provided configuration file.
+    General options:
+    -h --help         Print this help message.
+    -c --config-path  Path to the Configuration File (yaml format) [Default: ./DDS_ROUTER_CONFIGURATION.yaml].
+    -r --reload-time  Time period in milliseconds to reload configuration file. This is needed when FileWatcher functionality is not available (e.g. config file is a symbolic link).
+                        Value 0 does not reload file. [Default: 0].
+    -d --debug        Activate debug Logs. (Be aware that some logs may require specific CMAKE compilation options.compilation options)
+
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:88]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        -
+        -
+
+
+.. _user_manual_user_interface_help_argument:
+
+Help Argument
+^^^^^^^^^^^^^
+
+It shows the Manual information for the application.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:75]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+Set the :ref:`user_manual_user_interface_reload_timer` in **milliseconds**.
+
+
+.. _user_manual_user_interface_debug_argument:
+
+Debug Argument
+^^^^^^^^^^^^^^
+
+Activate ``INFO`` and ``DEBUG`` logs for the |ddsrouter| execution.
+For this argument to work, the |ddsrouter| must have been compiled in ``debug`` CMake build type, or compiled with
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:112]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+Configuration File
+------------------
+
+A |ddsrouter| **requires** one and only one *YAML* configuration file.
+Check section :ref:`user_manual_configuration` in order to know how to write this configuration file.
+
+This *YAML* configuration file must be passed as argument to the |ddsrouter| when executed.
+If configuration file is not defined as argument, the |ddsrouter| will be initialized with a file with
+default name ``DDS_ROUTER_CONFIGURATION.yaml`` that must be in the directory where the application has been executed.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:126]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+If the configuration file is not passed as argument, and it does not exist a file with the default configuration
+file name in the current directory, the application will fail.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:128]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+
+Reload Topics
+-------------
+
+The topics that the |ddsrouter| is routing could be changed in run time.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:134]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+Including topics in ``allowedlist`` in the configuration will create new :term:`Writers <DataWriter>` and
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:135]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+receive data.
+
+There exist two methods to reload the topics, an active and a passive one.
+Both methods work over the same configuration file which the |ddsrouter| has been initialized with.
+
+
+File Watcher
+^^^^^^^^^^^^
+
+A File Watcher is a process that runs in background and watch the |ddsrouter| configuration file.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:149]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+Every time that the file is changed, the OS sends a notification, and the File Watcher listens such notification
+and interact with the |ddsrouter| in order to reload the topics.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:151]: :+1:

File: docs/rst/user_manual/user_interface.rst [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+However, this method does not work properly in specific scenarios where the file being watched is not a real file but
+a link (e.g. Kubernetes executions).
+
+.. _user_manual_user_interface_reload_timer:
+
+Reload Timer
+^^^^^^^^^^^^
+
+A timer could be set in order to periodically reload the configuration file.
+The configuration file will be automatically reloaded with the period set.
```
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:164]: :+1:

File: docs/rst/developer_manual/installation/configuration/cmake_options.rst [hunk 15]
```diff
@@ -23,6 +23,11 @@ These options allow the developer to enable/disable certain *DDS Router* setting
         - Description
         - Possible values
         - Default
+    *   - :class:`CMAKE_BUILD_TYPE`
+        - CMake optimization build type
```
Comment (juanlofer-eprosima) [inline docs/rst/developer_manual/installation/configuration/cmake_options.rst:27]:
```suggestion
        - CMake optimization build type.
```

File: docs/rst/user_manual/user_interface.rst [hunk 16]
```diff
@@ -0,0 +1,172 @@
+.. include:: ../exports/alias.include
+
+.. _user_manual_user_interface:
+
+##############
+User Interface
+##############
+
+The |eddsrouter| is an application executed from command line.
```
Comment (juanlofer-eprosima) [inline docs/rst/user_manual/user_interface.rst:9]:
```suggestion
|eddsrouter| is an application executed from command line.
```

File: docs/rst/user_manual/user_interface.rst [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Source Dependency Libraries
+---------------------------
+
+The |eddsrouter| depends on |fastdds| ``fastrtps`` and ``fastcdr`` libraries.
```
Comment (juanlofer-eprosima) [inline docs/rst/user_manual/user_interface.rst:19]:
```suggestion
|eddsrouter| depends on |fastdds| ``fastrtps`` and ``fastcdr`` libraries.
```

File: docs/rst/user_manual/user_interface.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (juanlofer-eprosima) [reply docs/rst/user_manual/user_interface.rst]: Note still not visible

File: docs/rst/user_manual/user_interface.rst (hunk 16 above)
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst:9]: :+1:

File: docs/rst/user_manual/user_interface.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (jparisu) [reply docs/rst/user_manual/user_interface.rst]: true :cry:

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**